### PR TITLE
Users/ajya/create vp if not present

### PR DESCRIPTION
**Pull request by Ajay-MS**

### CODE REVIEW

File: Tasks/AzureRmWebAppDeployment/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -86,5 +86,10 @@
   "loc.messages.WebAppDoesntExist": "Webapp '%s' doesn't exist. Webapp should exist before deployment.",
   "loc.messages.EncodeNotSupported": "Detected file encoding of the file %s as %s. Variable substitution and Transformation is not supported with file encoding %s. Supported encodings are UTF-8 and UTF-16 LE.",
   "loc.messages.UnknownFileEncodeError": "Unable to detect encoding of the file %s (typeCode: %s). Supported encodings are UTF-8 and UTF-16 LE.",
-  "loc.messages.ShortFileBufferError": "File buffer is too short to detect encoding type : %s"
+  "loc.messages.ShortFileBufferError": "File buffer is too short to detect encoding type : %s",
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/Strings/resources.resjson/en-US/resources.resjson]: merge your changes with latest master and update the PR or make your changes on top of master that would be better.

File: Tasks/AzureRmWebAppDeployment/kuduutility.ts [hunk 2]
```diff
@@ -133,4 +133,45 @@ export async  function containsParamFile(webAppPackage: string ) {
...
+export async function createKuduPhysicalPath(publishingProfile, physicalPath: string) {
+    var defer = Q.defer<string>();
+    physicalPath = physicalPath.replace(/[\\]/g, "/");
+    var kuduPhysicalpathUrl = "https://" + publishingProfile.publishUrl + "/api/vfs/" + physicalPath + "/";
+    var basicAuthToken = 'Basic ' + new Buffer(publishingProfile.userName + ':' + publishingProfile.userPWD).toString('base64');
+    var headers = {
+        'Authorization': basicAuthToken,
+        'If-Match': "*"
+    };
+    tl.debug(tl.loc("RequestedURLforkuduphysicalpath", kuduPhysicalpathUrl))
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/kuduutility.ts]: remove localization here

File: Tasks/AzureRmWebAppDeployment/kuduutility.ts [hunk 3]
```diff
@@ -133,4 +133,45 @@ export async  function containsParamFile(webAppPackage: string ) {
...
+
+    httpObj.send('GET', kuduPhysicalpathUrl, null, headers, (error, response, body) => {
+        if (error) {
+            defer.reject(error);
+        }
+        else if (response.statusCode === 200 || response.statusCode === 201 || response.statusCode === 204) {
+            tl.debug(tl.loc("Physicalpathalreadyexists", physicalPath));
+            defer.resolve(tl.loc('Physicalpathalreadyexists'));
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/kuduutility.ts]: Is this message customer facing? If not remove localization.  I feel this message not required to shown to customer

File: Tasks/AzureRmWebAppDeployment/kuduutility.ts [hunk 4]
```diff
@@ -133,4 +133,45 @@ export async  function containsParamFile(webAppPackage: string ) {
     }
     tl.debug(tl.loc("Isparameterfilepresentinwebpackage0", isParamFilePresent));
     return isParamFilePresent;
+}
+
+export async function createKuduPhysicalPath(publishingProfile, physicalPath: string) {
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/kuduutility.ts]: rename to ensurePhysicalPathExists

File: Tasks/AzureRmWebAppDeployment/kuduutility.ts [hunk 5]
```diff
@@ -133,4 +133,45 @@ export async  function containsParamFile(webAppPackage: string ) {
...
+    var headers = {
+        'Authorization': basicAuthToken,
+        'If-Match': "*"
+    };
+    tl.debug(tl.loc("RequestedURLforkuduphysicalpath", kuduPhysicalpathUrl))
+
+    httpObj.send('GET', kuduPhysicalpathUrl, null, headers, (error, response, body) => {
+        if (error) {
+            defer.reject(error);
+        }
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/kuduutility.ts]: I believe you are first checking and then creating, Divide them into two functions to make function smaller.

**Comment by Ajay-MS** [comment]
Due to conflicts and migration of files creating new PR. closing this.

**Pull request closed without merging**